### ISSUE TITLE
Network monitor / debugger 

### ISSUE THREAD

**Issue by ducklin5** [godotengine/godot#28995]
TLDR; Some network debugging would be beneficial to the engine.
 
I'm going to start off by saying thank you to all the devs that make Godot what it is. It has so many great features and is the best game engine I've used to date!

Networking is a core part of many games and can be really hard to debug while writing code. Multiplayer components can only easily be debugged while running it so a networking section would be really helpful. I found myself using tools like clumsy and Netbalancer, but these don't expose the inner workings of the project. It would help to know (during runtime):
- incoming call frequency
- call validity (rather than all the red text in the console)
- packet size (this one's probably the most important for data usage)
- Network shaping (lag simulation, packet drops, etc...)

The last one is admittedly for a specific case, but I wasn't able to find a tool that would do networking shaping for just one instance of the game (most of the filtered packets by port).

**Comment by akien-mga** [comment]
CC @godotengine/network

**Comment by TheoXD** [comment]
For network stress-testing I can recommend GNS3, there you can create a real network topology and introduce packet drops.

**Comment by KoBeWi** [comment]
Network profiler was added in #31870

We are in process of moving proposals to [Godot proposals repository](https://github.com/godotengine/godot-proposals/). If you feel like the new profiler is lacking, open a new proposal there.
